### ISSUE TITLE
Doesn't compile when __cpp_modules is defined

### ISSUE THREAD

**Issue by sam20908** [boost-ext/ut#432]
## Expected Behavior
Compiles

## Actual Behavior
Does not

Background context: I was using this in my own project, and this bug affects the intellisense because of the error. I am using VSCode with C/C++ extension which was the extension that spotted the error

Spotted by https://github.com/microsoft/vscode-cpptools/issues/7150#issuecomment-796966568

## Steps to Reproduce the Problem

  1. Grab a copy of `ut.hpp`
  2. See error

I'm really sorry for the lack of detailed steps, as I can't afford to install GCC 11 and trying this myself. The comment I linked above describes the problem within `ut.hpp` and how it can be fixed.

## Specifications

  - Version: master branch
  - Platform: Windows 10
  - Subsystem: WSL 2

**Comment by michaelvlach** [comment]
I am facing the same issue with MSVC. After fixing some parsing issues like missing `ifdef` for the namespace endings the compilation fails with:
```
d:\A01\_work\7\s\src\vctools\Compiler\CxxFE\sl\p1\c\module\writer.cpp:5597: sorry: not yet implemented
../projects/ut/include/boost/ut.hpp(2297): fatal error C1001: Internal compiler error.
(compiler file 'd:\A01\_work\7\s\src\vctools\Compiler\CxxFE\sl\p1\c\module\utilities.h', line 34)
 To work around this problem, try simplifying or changing the program near the locations listed above.
If possible please provide a repro here: https://developercommunity.visualstudio.com
Please choose the Technical Support command on the Visual C++
 Help menu, or open the Technical Support help file for more information
```

**Comment by michaelvlach** [comment]
Addendum: Clang 11.01 also crashes when compiling with `__cpp_modules`. Unfortunately I am unable to pinpoint what is giving the compilers such trouble.

**Comment by sam20908** [comment]
Fixed with #437
